### DISCUSSION TITLE
Make Engine.get_main_loop() valid before _static_init()

### DISCUSSION THREAD

**Discussion by CsloudX** [godotengine/godot-proposals#7716] [Engine Core]
### Describe the project you are working on

NoGame App

### Describe the problem or limitation you are having in your project

![image](https://github.com/godotengine/godot-proposals/assets/35988767/4bfd8ec4-5bcc-430d-9d11-96a249a68341)
I'm so love Godot4's static variable and I want design a Helper class for auto set some information for Engine.get_main_loop().root, but failed. In _static_init function, Engine.get_main_loop() return null.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

IMO, it should alway can get Engine.get_main_loop() and Engine.get_main_loop().root.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

/

### If this enhancement will not be used often, can it be worked around with a few lines of script?

/

### Is there a reason why this should be core and not an add-on in the asset library?

/

**Comment by dalexeev** [comment]
I think it's probably bad practice to use `_static_init()` like this. Even if we fix the bug that the engine is not fully initialized by the time user code is executed, you still have little control over the order in which classes are loaded and their static constructors are called. In my opinion, `_static_init()` should only be used for more complex initialization of static variables.

**Reply by CsloudX** [reply]
No, I'm only care `Engine.get_main_loop()` and `Engine.get_main_loop().root` was valid in `_static_init()`. not care other static constructors

**Reply by CsloudX** [reply]
Hello, I'm test again. And I found in 4.1, it return null, in 4.2, it return a valid object.
So, was this a bug and was fixed 4.2?
